Objects/stringobject.c

```diff
@@ -1030,10 +1030,12 @@ string_concat(register PyStringObject *a, register PyObject *bb)
     register Py_ssize_t size;
     register PyStringObject *op;
     if (a->ob_type != bb->ob_type) {
+        char msgbuf[256];
+        sprintf(msgbuf, "the first string is '%.200s' while the second is '%.200s': "\
```

Comment:
Is it possible to put the *type* rather than (I think) the value here? So it says something like "the LHS is unicode the RHS bytes" or whatever?

Reply:
This extracts the type using `Py_TYPE(bb)->tp_name)` so it is either 'str' or `unicode` or `bytes` in the message. Am Am I understanding you clearly?

Reply:
I'm an idiot!

Reply:
No you arent :)